include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -15,6 +15,22 @@
 
 include "mlir/IR/SymbolInterfaces.td"
 include "mlir/IR/OpBase.td"
+def ModuleLike : OpInterface<"ModuleLike", []> {
```

Comment:
It is weird that this is implemented by `FIRRTLOp`, `HWOp`, and `MSFTOp`. Those aren't really modules.

This is more `HasName` than `ModuleLike` in that respect.

WDYT?

Reply:
How is it implemented by FIRRTLOp?  It's implemented by FIRRTLModuleLike base of of F*ModuleOp

Reply:
I can't read the ODS. 😢 Ignore this comment.